src/Tester/GenericGrainTests.cs

```diff
@@ -299,7 +299,7 @@ public static void MyClassCleanup()
             await grain.SetB(b);
 
             Task<string> stringPromise = grain.GetAxB();
-            Assert.AreEqual(expected, stringPromise.Result);
```

Comment:
We definitely do NOT want to change this test.
This test was the only test in our CI that broke due to the wrong usage of `ExecuteSynchronosuly` in https://github.com/dotnet/orleans/pull/853.

There is nothing wrong or illegal about mixing up `await` and `.Wait` **on the client**. Indeed, it is a bad style, and people should try to avoid it (either all `await` or all blocking Wait), but TPL does allow it, and Orleans cannot prevent its customers from doing so **on the client** and thus we should support it and also test it.